bindings/python/mata.pxd

```diff
@@ -245,6 +245,7 @@ cdef extern from "mata/nfa.hh" namespace "Mata::Nfa":
 
 cdef extern from "mata/noodlify.hh" namespace "Mata::Nfa::SegNfa":
     cdef vector[CNfa] noodlify(CNfa&, Symbol, bool)
+    cdef vector[CNfa] noodlify_for_equation(const vector[const CNfa*]&, CNfa&, bool)
```

Comment:
I suggest extracting type `vector[CNfa*]` to isolate type for better readability (maybe `cnfa_vector`?`

Also, I believe `const` are not really that much necessary in the Cython, it might help with performance (but I doubt it will).